topics/go/algorithms/data/bst/bst.go

```diff
@@ -0,0 +1,113 @@
+package bst
+
+import "fmt"
+
+// TODO This is currently a draft...
+
+// Node represents the data being stored.
+type Node struct {
+	Data  int
+	left  *Node
+	right *Node
+}
+
+// BST represents a binary search tree.
+type BST struct {
+	root *Node
+}
+
+// New creates a new BST.
+func New() *BST {
+	return &BST{
+		root: nil,
+	}
+}
+
+// Insert an integer into the BST.
+func (bst *BST) Insert(value int) {
+
+	// Create a new node.
+	n := &Node{
+		Data:  value,
```

Comment:
Don't use pointer semantic contruction to a variable.

Reply:
Why?

Reply:
For readability. The & shows sharing and you don't want to lose that information.

We make things easy to understand, not easy to do.